### PR TITLE
Proof of Account verifier

### PR DESCRIPTION
**Pull request by xqft**
This PR:
- adds a first version of the merkle verifier for Mina accounts
- changes the state verifier to include the state's merkle root in its public inputs

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -666,6 +689,8 @@ build_all_ffi_linux: ## Build all FFIs for Linux
 	@$(MAKE) build_merkle_tree_linux
 	@$(MAKE) build_halo2_ipa_linux
 	@$(MAKE) build_halo2_kzg_linux
+	@$(MAKE) build_mina_linxu
```
Comment (gabrielbosio) [inline Makefile]:
```suggestion
	@$(MAKE) build_mina_linux
```

File: batcher/aligned-batcher/.env.dev [hunk 2]
```diff
@@ -2,7 +2,7 @@ AWS_SECRET_ACCESS_KEY=test
 AWS_REGION=us-east-2
 AWS_ACCESS_KEY_ID=test
 AWS_BUCKET_NAME=aligned.storage
-STORAGE_ENDPOINT=http://localhost:4566/aligned.storage
+STORAGE_ENDPOINT="https://mina-bridge.s3.us-west-2.amazonaws.com"
```
Comment (gabrielbosio) [inline batcher/aligned-batcher/.env.dev]: I think we should leave this var unchanged and set it to the value we want in `.env`. If not, we should change the other vars to the corresponding values (except the secrets).

File: operator/mina_account/lib/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        Err(err) => {
+            eprintln!("Failed to parse merkle proof: {}", err);
+            return false;
+        }
+    };
+
+    // TODO(xqft): when the needed account GraphQL query is done, do:
+    // 1. send encoded account as part of the proof
+    // 2. define account_id from encoded account.
+    // 2. assert keccak256(account_id) == account_id_hash
```
Comment (gabrielbosio) [inline operator/mina_account/lib/src/lib.rs]:
```suggestion
    // 3. assert keccak256(account_id) == account_id_hash
```

File: Makefile (hunk 1 above)
Comment (xqft) [reply Makefile]: good catch!!

File: batcher/aligned-batcher/.env.dev (hunk 2 above)
Comment (xqft) [reply batcher/aligned-batcher/.env.dev]: I agree with leaving this unchanged!

**Review by gabrielbosio — APPROVED** [review]

**Pull request merged**